plone/app/textfield/transform.py

```diff
@@ -67,3 +79,34 @@ def __call__(self, value, mimeType):
             # log the traceback of the original exception
             LOG.error("Transform exception", exc_info=True)
             raise TransformError('Error during transformation', e)
+
+    def check_referenced_images(self, target_mimetype, cache_obj):
+        # referenced image scale urls get outdated if the images are modified.
+        # purging the transform cache updates those urls.
+        cache = Cache(cache_obj, context=self.context)
+        data = cache.getCache(target_mimetype)
+
+        if data is None:
+            # not cached ... return
+            return
+
+        # lookup referenced images and check modification time
+        try:
+            ol = getOutgoingLinks(self.context)
+        except KeyError:
+            # context might not implement IIntId (eg. portal)
+            return
+
+        # get the original save time from the cached data dict
+        orig_time = getattr(cache_obj, cache._id).values()[0][0]
+
+        for ref_link in ol:
+            # only lookup image references
+            if IImage not in ref_link.to_interfaces_flattened:
+                continue
+            # XXX: not sure if it is a potential performance problem
+            # looking up the image object
+            if ref_link.to_object.modified() > orig_time:
```

Comment:
I am not sure whats better, waking up the object or query the catalog. Without performance monitoring this is difficult to say.